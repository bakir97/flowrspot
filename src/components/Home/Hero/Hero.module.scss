.container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 500px;
  background-image: url("../../../assets/images/pl-hero.png");
  background-size: cover;
}
