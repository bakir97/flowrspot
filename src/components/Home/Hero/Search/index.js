import Search from "./Search";
import { connect } from "react-redux";
import { searchFlowers } from "../../../../redux/flowers/actions";
const mapStateToProps = state => ({});

const mapDispatchToProps = {
  searchFlowers
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Search);
