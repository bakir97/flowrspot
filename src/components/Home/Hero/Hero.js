import React, { Component } from "react";
import styles from "./Hero.module.scss";
import Search from "./Search";
import HeroText from "./HeroText";
export default class Hero extends Component {
  handleSearch = search => {
    this.props.handleSearch(search);
  };
  render() {
    return (
      <div className={styles.container}>
        <HeroText />
        <Search handleSearch={this.handleSearch} />
      </div>
    );
  }
}
