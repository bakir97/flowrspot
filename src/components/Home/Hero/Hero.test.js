import React from "react";
import { shallow } from "enzyme";
import Hero from "./Hero";
describe("Hero", () => {
  it("should render correctly", () => {
    const component = shallow(<Hero />);
    expect(component).toMatchSnapshot();
  });
  it("should call handleSearch", () => {
    const handleSearch = jest.fn();
    const component = shallow(<Hero handleSearch={handleSearch} />);
    component.instance().handleSearch("test");
    expect(handleSearch).toHaveBeenCalledWith("test");
  });
});
