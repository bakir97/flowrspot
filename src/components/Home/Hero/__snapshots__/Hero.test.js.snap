// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should render correctly 1`] = `
<div
  className="container"
>
  <HeroText />
  <Connect(Search)
    handleSearch={[Function]}
  />
</div>
`;
